src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HKDF.cs

```diff
@@ -134,35 +134,66 @@ private static void Expand(HashAlgorithmName hashAlgorithmName, int hashLength,
             Span<byte> t = Span<byte>.Empty;
             Span<byte> remainingOutput = output;
 
-            using (IncrementalHash hmac = IncrementalHash.CreateHMAC(hashAlgorithmName, prk))
+            ReadOnlySpan<byte> infoBuffer = stackalloc byte[0];
```

Comment:
While this looks like .NET Core 2.0 Jeremy code, I feel like .NET 6 Jeremy code would just always make the stackalloc.

```C#
Span<byte> tempInfoBuffer = stackalloc byte[64];
ReadOnlySpan<byte> infoBuffer = tempInfoBuffer;
byte[]? rentedTempInfoBuffer = null;

if (overlaps)
{
    if (info.Length > tempInfoBuffer.Length)
    {
        rented = Rent();
        tempInfoBuffer = rented;
    }

    tempInfoBuffer = tempInfoBuffer.Slice(0, info.Length);
    info.CopyTo(tempInfoBuffer);
}
else
{
    infoBuffer = info;
}
```

Reply:
Hm. That's fair. I like the readability of this more.